File: src/nodes/eof.cpp
#include "nodes/eof.hpp"
#include <iostream>

using namespace nodes;

NodeEndOfFile::NodeEndOfFile(const utils::Position &position)
    : Node(position) {

}

void NodeEndOfFile::show(std::string &indent, char end) const {
    std::cout << indent << "NodeEndOfFile" << end;
}

NodeKind NodeEndOfFile::getKind() const {
    return NodeKind::EndOfFile;
}

void NodeEndOfFile::emit(core::Emitter &) const {
    
}

void NodeEndOfFile::setId() {
    
}

utils::OptimizedResult NodeEndOfFile::optimize(core::Optimizer &) {
    return utils::OptimizedResult(utils::OptimizedFlag::None);
}

File: src/nodes/eof.rs
use crate::utils::position::Position;

#[derive(Clone)]
pub struct NodeEndOfFile {
    position: Position,
}

impl NodeEndOfFile {
    pub fn init(position: Position) -> NodeEndOfFile {
        NodeEndOfFile { position }
    }
    
    pub fn get_position(&self) -> &Position {
        &self.position
    }
    
    pub fn show(&self, indent: &mut String, end: char) {
        print!("{indent}NodeEndOfFile{end}");
    }
}